browserApp/info_view.py

```diff
@@ -1,35 +1,152 @@
+        self.list_layout.addWidget(self.size_header, 0, 1)
+        self.list_layout.addWidget(self.type_header, 0, 2)
+        self.list_layout.addWidget(self.created_header, 0, 3)
+
+        self.list_view_button = QtWidgets.QPushButton("List View")
+        self.icons_view_button = QtWidgets.QPushButton("Icon View")
+        self.top_bar_layout = QtWidgets.QHBoxLayout()
+        self.top_bar_layout.addWidget(self.list_view_button)
+        self.top_bar_layout.addWidget(self.icons_view_button)
+
+        self.list_view_button.clicked.connect(self.setup_list_view)
+        self.icons_view_button.clicked.connect(self.setup_icons_view)
+
+        self.stacked_pages = QtWidgets.QStackedWidget()
+        self.stacked_pages.addWidget(self.list_table)
+        self.stacked_pages.addWidget(self.icons_widget)
+
+        self.stacked_pages.setCurrentWidget(self.list_table)
 
         layout.addSpacing(1)
-        layout.addWidget(self.table)
+        layout.addLayout(self.top_bar_layout)
+        layout.addWidget(self.stacked_pages)
         layout.addSpacing(1)
 
+        self.column_index = 0
+        self.row_index = 0
 
         self.setLayout(layout)
 
     def populate(self, data):
-        print("Do something useful with: " + str(data))
+        # print("Do something useful with: " + str(data))
 
-        # Clear the table
-        self.table.clear()
+        # Clear the list_table
+        self.clear_layout(self.list_layout)
+        self.stacked_pages.setCurrentWidget(self.list_table)
 
-        # Update the table with name : value from the get_info() dict
+        # If item clicked on is a file, display info in list_table. Else, display buttons with file names.
         meta = data.get_info()
-        for k, v in meta.items():
-            item = QtWidgets.QTreeWidgetItem([k,v])
-            self.table.addTopLevelItem(item)
+        if 'type' in meta:
+            # File:
+            if meta['type'] == 'File':
+                pass
+                # for k, v in meta.items():
+                #     item = QtWidgets.QTreeWidgetItem([k, v])
+                #     self.details_list_table.addTopLevelItem(item)
+
+            # Directory:
+            elif meta['type'] == 'Dir':
+                self.stacked_pages.setCurrentWidget(self.icons_widget)
+                self.icons_widget.setLayout(self.icons_layout)
+
+                self.clear_layout(self.icons_layout)
+                self.icons_layout.setAlignment(QtCore.Qt.AlignTop)
+
+                # Request addition of 'file_name' to data model
+                if 'full_path' in meta:
+                    dir_path = meta['full_path']
+                    files = [f for f in listdir(dir_path) if isfile(join(dir_path, f))]
+                    buttons_list = []
+                    print(files)
+
+                    for file in files:
+                        file_name = file.split('.')[0]
+                        exec("%s_button = QtWidgets.QPushButton('%s')" % (file_name, file_name))
```

Comment:
this looks hacky - I bet there is a better way to do this - maybe keep a dict of button objects?
self.buttons[file_name] = QtWidgets.QPushButton(file_name)